ntfsutils/fs.py

```diff
@@ -90,7 +91,12 @@ def getfileinfo(path):
     Return information for the file at the given path. This is going to be a
     struct of type BY_HANDLE_FILE_INFORMATION.
     """
-    hfile = CreateFile(path, GENERIC_READ, FILE_SHARE_READ, None, OPEN_EXISTING, 0, None)
+    if os.path.isdir(path):
```

Comment:
So I'm a little bit concerned about [TOCTOU](https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use) races here -- i.e. if something comes in and changes the path in between the `os.path.isdir` call and the `CreateFile` below. Would it be worth doing something like trying to call `CreateFile` without this flag first, then if it fails (and you can see that it's a directory) to try and call it with this flag?

Reply:
I totally agree. Updated the code to try once again with FILE_FLAG_BACKUP_SEMANTICS flag. Please take a look.